perllib/FixMyStreet/App/Controller/Admin/AreaStats.pm

```diff
@@ -0,0 +1,208 @@
+    my ($self, $c) = @_;
+
+    my $date = DateTime->now->subtract(days => 30);
+    my $area_id = $c->stash->{area_id};
+    my $area = mySociety::MaPit::call('area', $area_id );
+    my $user = $c->user;
+
+    $c->forward('/admin/fetch_contacts');
+
+    $c->stash->{template} = 'admin/areastats/area.html';
+    if ($area->{name}) {
+        $c->stash->{area} = $area;
+
+        my $dtf = $c->model('DB')->storage->datetime_parser;
+        my $time = $dtf->format_datetime($date);
+
+        my $params = {
+            'problem.areas' => { like => "%,$area_id,%" },
+            'me.confirmed' => { '>=', $time },
+        };
+
+        my $comments = $c->model('DB::Comment')->search(
+            $params,
+            {
+                group_by => [ 'problem_state' ],
+                select   => [ 'problem_state', { count => 'me.id' } ],
+                as       => [ qw/state state_count/ ],
+                join     => 'problem'
+            }
+        );
+
+        $params = {
+            areas => { like => "%,$area_id,%" },
+        };
+
+        my $problems = $c->model('DB::Problem')->search(
+            $params,
+            {
+                group_by => [ 'category', 'state' ],
+                select   => [ 'category', 'state', { count => 'me.id' } ],
+                as       => [ qw/category state state_count/ ],
+            }
+        );
+
+        my %by_category = map { $_->category => {} } $c->stash->{contacts}->all;
+        my %recent_by_category = map { $_->category => 0 } $c->stash->{contacts}->all;
+
+        my $state_map = {};
+
+        $state_map->{$_} = 'open' foreach FixMyStreet::DB::Result::Problem->open_states;
+        $state_map->{$_} = 'closed' foreach FixMyStreet::DB::Result::Problem->closed_states;
+        $state_map->{$_} = 'fixed' foreach FixMyStreet::DB::Result::Problem->fixed_states;
+        $state_map->{$_} = 'scheduled' foreach ('planned', 'action scheduled');
+
+        for my $p ($problems->all) {
+            my $meta_state = $state_map->{$p->state};
+            $by_category{$p->category}->{$meta_state} += $p->get_column('state_count');
+        }
+
+        $params->{'me.confirmed'} = { '>=', $time };
+
+        my $recent_problems = $c->model('DB::Problem')->search(
+            $params,
+            {
+                group_by => [ 'category' ],
+                select   => [ 'category', { count => 'me.id' } ],
+                as       => [ qw/category category_count/ ],
+            }
+        );
+
+        for my $p ($recent_problems->all) {
+            $recent_by_category{$p->category} += $p->get_column('category_count');
+        }
+
+
+        $c->stash->{$_} = 0 for values %$state_map;
+        $c->stash->{open} = $c->model('DB::Problem')->search(
+            {
+                areas => { like => "%,$area_id,%" },
+                confirmed => { '>=' => DateTime::Format::W3CDTF->format_datetime($date) },
+            }
+        )->count;
+
+        for my $comment ($comments->all) {
```

Comment:
A comment `problem_state` is set (most of) the time with an update, regardless of if there's been a state change or not; so I don't think it can be used as is as a statistic for the number of issues scheduled/closed/fixed in the past month :-/ You could perhaps go on staff users only? But not even then I imagine. Not sure what to suggest here, something to discuss.